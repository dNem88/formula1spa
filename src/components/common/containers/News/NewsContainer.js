import React from 'react'
import styles from './NewsContainer.module.css'
import {useNavigate} from 'react-router-dom'

function NewsContainer({id, path, type, children}) {

    let navigate = useNavigate();

    function clickHandler(e) {
        navigate(`/${path}/${id}`)
    }

    return (
        <div onClick={clickHandler} id={id} className={styles[`main-container-${type}`]}>
            {children}
        </div>
    )
}

export default NewsContainer