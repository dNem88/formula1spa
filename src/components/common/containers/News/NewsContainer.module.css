.main-container-vertical{
    box-sizing: border-box;
    width: 23vw;
    height: 38vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    cursor: pointer;
}

.main-container-vertical:hover > :nth-child(1){
    height: 47%;
    transition: height .35s ease;
    background-color: var(--color-dark-blue);
    
}
.main-container-vertical:hover > :nth-child(2){
    height: 53%;
    transition: height .35s ease;
    border-bottom-color: var(--color-red);
    border-right-color: var(--color-red);
    background-color: var(--color-dark-blue);
    border-bottom-right-radius: 1rem;
    color: white;
}


@media only screen and (max-width: 750px) {
    .main-container-vertical{
        width: 47vw;
        height: 50vh;
    }
   
}
@media only screen and (max-width: 800px) and (orientation: landscape){
    .main-container-vertical{
        width: 45vw;
        height: 80vh;
    }
    
}