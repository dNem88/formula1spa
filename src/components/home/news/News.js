import React, {useRef} from 'react';
import styles from './news.module.css'

function News(props) {
    console.log(props.hoverChange)
    return (
        <div className={styles['main-container']}>
            <div className={styles['image-container']}>
                <img id={styles['news-image']} alt={'some image'} src={props.imageUrl}/>
            </div>
            <div className={props.hoverChange === true ? styles['dark-container'] : styles.container}>
                <p className={styles['news-par']}>NEWS</p>
                <p className={styles['content-par']}>{props.content}</p>
            </div>   
        </div>
    )
}

export default News