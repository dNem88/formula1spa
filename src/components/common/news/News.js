import React from 'react';
import styles from './news.module.css'
import NewsContainer from '../containers/News/NewsContainer';

function News({id, imageUrl, hoverChange, title}) {
  

    return (
        <NewsContainer id={id} path={'news'} type={'vertical'}>
            <div className={styles['image-container']}>
                <img id={styles['news-image']} alt={'image-from-newsapi'} src={imageUrl} loading='lazy'/>
            </div>
            <div className={hoverChange === true ? styles['dark-container'] : styles.container}>
                <p className={styles['news-par']}>NEWS</p>
                <p className={styles['content-par']}>{title}</p>
            </div>   
        </NewsContainer>
        
    )
}

export default News