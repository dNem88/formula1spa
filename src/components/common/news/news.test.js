import React from 'react'
import News from './news'
import renderer from 'react-test-renderer'
import {MemoryRouter} from 'react-router-dom'
describe('Testing news Component', () => {
    it('Testing News element', () => {
        let news = renderer.create(
            <MemoryRouter>
                <News hoverChange={false} id={'test-id'} imageUrl={'test-url'} title={'test-title'}/>
            </MemoryRouter>
        ).toJSON();
        expect(news).toMatchSnapshot()
        expect(news.type).toBe('div')
        expect(news.props.className).toBe('main-container')
        expect(news.props.id).toBe('test-id')
        expect(news.children[0].type).toBe('div')
        expect(news.children[0].props.className).toBe('image-container')
        expect(news.children[1].type).toBe('div')
        expect(news.children[1].props.className).toBe('container')
        expect(news.children[0].children[0].type).toBe('img')
        expect(news.children[0].children[0].props.id).toBe('news-image')
        expect(news.children[0].children[0].props.src).toBe('test-url')
        expect(news.children[1].children[0].type).toBe('p')
        expect(news.children[1].children[0].props.className).toBe('news-par')
        expect(news.children[1].children[0].children[0]).toBe('NEWS')
        expect(news.children[1].children[1].type).toBe('p')
        expect(news.children[1].children[1].props.className).toBe('content-par')
        expect(news.children[1].children[1].children[0]).toBe('test-title')
    })
    it('Testing News elements with changed className', () => {
        let newsTwo = renderer.create(
            <MemoryRouter>
                <News hoverChange={true} id={'test-id'} imageUrl={'test-url'} title={'test-title'}/>
            </MemoryRouter>
        ).toJSON();
        expect(newsTwo).toMatchSnapshot()
        expect(newsTwo).toMatchSnapshot()
        expect(newsTwo.type).toBe('div')
        expect(newsTwo.props.className).toBe('main-container')
        expect(newsTwo.props.id).toBe('test-id')
        expect(newsTwo.children[0].type).toBe('div')
        expect(newsTwo.children[0].props.className).toBe('image-container')
        expect(newsTwo.children[1].type).toBe('div')
        expect(newsTwo.children[1].props.className).toBe('dark-container')
        expect(newsTwo.children[0].children[0].type).toBe('img')
        expect(newsTwo.children[0].children[0].props.id).toBe('news-image')
        expect(newsTwo.children[0].children[0].props.src).toBe('test-url')
        expect(newsTwo.children[1].children[0].type).toBe('p')
        expect(newsTwo.children[1].children[0].props.className).toBe('news-par')
        expect(newsTwo.children[1].children[0].children[0]).toBe('NEWS')
        expect(newsTwo.children[1].children[1].type).toBe('p')
        expect(newsTwo.children[1].children[1].props.className).toBe('content-par')
        expect(newsTwo.children[1].children[1].children[0]).toBe('test-title')
    })
})