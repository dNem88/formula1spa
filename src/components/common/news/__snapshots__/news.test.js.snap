// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing news Component Testing News element 1`] = `
<div
  className="main-container"
  id="test-id"
  onClick={[Function]}
>
  <div
    className="image-container"
  >
    <img
      alt="image-from-newsapi"
      id="news-image"
      loading="lazy"
      src="test-url"
    />
  </div>
  <div
    className="container"
  >
    <p
      className="news-par"
    >
      NEWS
    </p>
    <p
      className="content-par"
    >
      test-title
    </p>
  </div>
</div>
`;

exports[`Testing news Component Testing News elements with changed className 1`] = `
<div
  className="main-container"
  id="test-id"
  onClick={[Function]}
>
  <div
    className="image-container"
  >
    <img
      alt="image-from-newsapi"
      id="news-image"
      loading="lazy"
      src="test-url"
    />
  </div>
  <div
    className="dark-container"
  >
    <p
      className="news-par"
    >
      NEWS
    </p>
    <p
      className="content-par"
    >
      test-title
    </p>
  </div>
</div>
`;

exports[`Testing news Component Testing News elements with changed className 2`] = `
<div
  className="main-container"
  id="test-id"
  onClick={[Function]}
>
  <div
    className="image-container"
  >
    <img
      alt="image-from-newsapi"
      id="news-image"
      loading="lazy"
      src="test-url"
    />
  </div>
  <div
    className="dark-container"
  >
    <p
      className="news-par"
    >
      NEWS
    </p>
    <p
      className="content-par"
    >
      test-title
    </p>
  </div>
</div>
`;
